scifight/utils.py

```diff
@@ -0,0 +1,39 @@
+import re
+
+
+def shorten_text(text, maxchars, ellipsis='...'):
+    """ Returns first few words from the `text`, but no more than `maxchars`, and appends
+        an `ellipsis`. This function works only on plain text (no HTML!), removes any leading
+        and trailing spaces, and squashes any number of consequent whitespaces (including
+        newlines) into a single space.
+
+        :param text: Input string, possibly quite long.
+        :param maxchars: Maximum length or result string.
+        :param ellipsis: Trailing sign to indicate implied continuation.
+        :return: Shortened substring with squashed whitespaces.
+    """
+    if not text or maxchars <= 0:
+        return None
+    if not ellipsis:
+        ellipsis = ''
+
+    counter = 0
+    for i in range(len(text)):
```

Comment:
You better rewrite this loop as `for ch in text`, because string indexing isn't a cheap operation.